### PR TITLE
perf: change `ReplayPayload` structure

### PR DESCRIPTION
**Pull request by anonrig**
Reduces JSON serialization

### CODE REVIEW

File: rust_snuba/src/processors/replays.rs [hunk 1]
```diff
@@ -209,9 +212,15 @@ pub fn deserialize_message(
     Ok((rows, replay_message.start_time))
 }
 
+#[derive(Debug, Deserialize)]
+enum RawReplayPayload {
```
Comment (mitsuhiko) [inline rust_snuba/src/processors/replays.rs:216]: This needs a `#[serde(untagged)]` annotation.

**Review by mitsuhiko — DISMISSED** [review]

File: rust_snuba/src/processors/replays.rs (hunk 1 above)
Comment (anonrig) [reply rust_snuba/src/processors/replays.rs:217]: Done 👍

**Comment by john-z-yang** [comment]
Hey @anonrig, I'm not really familiar with the replays consumers.

I was wondering if you can provide some additional details to what this change does? It seems like the `payload` field use to be bytes and now we are planning to parse them into `ReplayPayload`?

Thanks

**Comment by anonrig** [comment]
Hey @john-z-yang,

We were using an unoptimized data structure, this PR introduces a non-breaking way of changing it. Sentry changes are done on https://github.com/getsentry/sentry/pull/69916

**Comment by john-z-yang** [comment]
@anonrig Thanks for the additional context.

The snuba consumers validates its messages using the schema defined in [sentry-kafka-schema](https://github.com/getsentry/sentry-kafka-schemas/blob/main/schemas/ingest-replay-events.v1.schema.json) repository. Messages that do not fit the schema will get dropped.

Please update the schema definition there, create a release (instructions [here](https://github.com/getsentry/sentry-kafka-schemas?tab=readme-ov-file#release-process-and-development-install)), and then update the cargo dependency with the new release please.

In your PR to sentry-kafka-schema, do you mind also adding some example messages ? I think that will help me understand what the unoptimized/optimized data structure looks like. Snuba also have automations that generates snap shots for consumers to prevent regressions.

Feel free to dm me with questions!

**Comment by anonrig** [comment]
@john-z-yang Thanks! Opened a PR https://github.com/getsentry/sentry-kafka-schemas/pull/265

**Pull request merged**